test/ductile/document_test.clj

```diff
@@ -149,440 +150,448 @@
+                 :max 2.0
+                 :avg 1.0
+                 :sum 3.0}
+                (:price_stats aggs-3))))))))
 
 (deftest ^:integration delete-by-query-test
-  (for-each-es-version
-   "delete-by-query should tigger a proper _delete_by_query request"
-   #(es-index/delete! conn "test_index")
-   (let [sample-docs-1 (mapv #(cond-> {:_index "test_index-1"
-                                       :foo (if (< % 5)
-                                              :bar1
-                                              :bar2)
-                                       :_id %}
-                                (= 5 version) (assoc :_type "doc-type"))
-                             (range 10))
-         sample-docs-2 (mapv #(cond-> {:_index "test_index-2"
-                                       :foo (if (< % 5)
-                                              :bar1
-                                              :bar2)
-                                       :_id %}
-                                (= 5 version) (assoc :_type "doc-type"))
-                             (range 10))
-         q-term (query/term :foo :bar2)
-         q-ids-1 (query/ids ["0" "1" "2"])
-         q-ids-2 (query/ids ["3" "4"])]
-     (es-index/delete! conn "test_index")
-     (es-index/create! conn "test_index" {})
-     (sut/bulk-create-doc conn sample-docs-1 {:refresh "true"})
-     (sut/bulk-create-doc conn sample-docs-2 {:refresh "true"})
-     (is (= 5
-            (:deleted (sut/delete-by-query conn
-                                           ["test_index-1"]
-                                           q-term
-                                           {:wait_for_completion true
-                                            :refresh "true"})))
-         "delete-by-query should delete all documents that match a query for given index")
-     (is (= 6
-            (:deleted (sut/delete-by-query conn
-                                           ["test_index-1", "test_index-2"]
-                                           q-ids-1
-                                           {:wait_for_completion true
-                                            :refresh "true"})))
-         "delete-by-query should properly apply deletion on all given indices")
-     (is (seq (:task (sut/delete-by-query conn
-                                          ["test_index-1", "test_index-2"]
-                                          q-ids-2
-                                          {:wait_for_completion false
-                                           :refresh "true"})))
-         "delete-by-query with wait-for-completion? set to false should directly return an answer before deletion with a task id"))))
+  (let [indexname (str "test_index" (UUID/randomUUID))
+        indexname1 (str indexname "-1")
+        indexname2 (str indexname "-2")]
+    (for-each-es-version
+     "delete-by-query should trigger a proper _delete_by_query request"
+     #(es-index/delete! conn indexname)
```

Comment:
We should add a wildward to ensure that "indexname-1" and "indexname-2" are actually cleaned if the `delete-by-query` failed in the tests (I relied on it, but it's better to add the wildcard too):
```suggestion
     #(es-index/delete! conn (str indexname "*")
```

Reply:
Makes sense. I was wondering what kind of magic made those indices get deleted.